Terminal.Gui/Application.cs

```diff
@@ -1469,11 +1478,11 @@ internal static void OnMouseEvent (MouseEventEventArgs a)
 
             var me = new MouseEvent
             {
-                X = screenX,
-                Y = screenY,
+                X = a.MouseEvent.X - screen.X,
```

Comment:
At some point (incoming pun alert), it would be nice for any type we have that has a location to keep it as a Point struct, for consistency (or a Rectangle, if that's more relevant for the given type). X and Y can still be exposed for convenience, of course, and just proxy to the properties on the Point.

Using Point or Rectangle, as appropriate for each type, also gives nice stuff like the Offset method and others which can make quite a few spots in our code a lot cleaner, clearer, and more consistent, too. Plus it helps eliminate the work of having to debug that math for every new method we add.

Reply:
Agree!